src/Catty/ViewController/Stage/StagePresenterViewController.m

```diff
@@ -435,7 +459,19 @@ - (void)showHideAxisAction:(UIButton*)sender
 - (void)manageAspectRatioAction:(UIButton *)sender
 {
     self.stage.scaleMode = self.stage.scaleMode == SKSceneScaleModeAspectFit ? SKSceneScaleModeFill : SKSceneScaleModeAspectFit;
-    self.project.header.screenMode = [self.project.header.screenMode isEqualToString:kCatrobatHeaderScreenModeStretch] ? kCatrobatHeaderScreenModeMaximize :kCatrobatHeaderScreenModeStretch;
+    if ([self.project.header.screenMode isEqualToString:kCatrobatHeaderScreenModeStretch]) {
+         self.project.header.screenMode = kCatrobatHeaderScreenModeMaximize;
+         [self.menuMaximizedMinimizedLabel setTitle:kLocalizedMinimize forState:UIControlStateNormal];
+        [self setupButtonWithButton:self.menuAspectRatioButton
+                    ImageNameNormal:[UIImage imageNamed:@"stage_dialog_button_aspect_ratio_close"]
+                        andSelector:@selector(manageAspectRatioAction:)];
+     } else {
+         self.project.header.screenMode = kCatrobatHeaderScreenModeStretch;
+         [self.menuMaximizedMinimizedLabel setTitle:kLocalizedMaximize forState:UIControlStateNormal];
+         [self setupButtonWithButton:self.menuAspectRatioButton
+                     ImageNameNormal:[UIImage imageNamed:@"stage_dialog_button_aspect_ratio"]
+                         andSelector:@selector(manageAspectRatioAction:)];
+     }
```

Comment:
@zwerg4 I guess there is a little bug, `self.menuMaximizedMinimizedLabel setTitle` gets only called if you click on the aspectRationButton. So if you maximize the view and the restart the game, the text of the label does not get updated. You could for example check the `project.header.screenMode` also in the `viewWillAppear` method I'd say 🤔

Reply:
Great code review, thanks!